### PR TITLE
doc: update the upgrade known issue

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
Related issue: https://github.com/harvester/harvester/issues/4526

### CODE REVIEW

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 1]
```diff
@@ -398,3 +398,76 @@ If an upgrade is stuck in an `Upgrading System Service` state for an extended pe
...
+
+date: "2023-09-13T08:31:42Z"
+state:
+    # we are missing `label` here.
+    active:
+        source: dir:///tmp/tmp.01deNrXNEC
+        label: COS_ACTIVE
+        fs: ext2
+    passive: null
+```
```
Comment (LucasSaintarbor) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:446]:
```suggestion
The incomplete `state.yaml` will look like the following.

```yaml
# Autogenerated file by elemental client, do not edit

date: "2023-09-13T08:31:42Z"
state:
    # we are missing `label` here.
    active:
        source: dir:///tmp/tmp.01deNrXNEC
        label: COS_ACTIVE
        fs: ext2
    passive: null
```
```

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 2]
```diff
@@ -398,3 +398,76 @@ If an upgrade is stuck in an `Upgrading System Service` state for an extended pe
...
+
+The simple workaround is removing this incomplete `state.yaml` file. (The post-draining will retry every 10 minutes).
```
Comment (LucasSaintarbor) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:448]:
```suggestion
Remove this incomplete `state.yaml` file to work around this issue. (The post-draining will retry every 10 minutes).

```

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 3]
```diff
@@ -398,3 +398,76 @@ If an upgrade is stuck in an `Upgrading System Service` state for an extended pe
...
+  $ rm -f /run/initramfs/cos-state/state.yaml
+  ```
+
+1. remount the `state` partition to RO
+
+  ```
+  $ mount -o remount,ro /run/initramfs/cos-state
+  ```
+
+Then, you should pass post-draining with the next retry.
```
Comment (LucasSaintarbor) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:467]:
```suggestion
After performing the steps above, you should pass post-draining with the next retry.
```

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 4]
```diff
@@ -398,3 +398,76 @@ If an upgrade is stuck in an `Upgrading System Service` state for an extended pe
...
+        fs: ext2
+    passive: null
+```
+
+The simple workaround is removing this incomplete `state.yaml` file. (The post-draining will retry every 10 minutes).
+1. remount the `state` partition to RW
+
+  ```
+  $ mount -o remount,rw /run/initramfs/cos-state
+  ```
```
Comment (LucasSaintarbor) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:453]:
```suggestion
1. Remount the `state` partition to RW.

  ```bash
  $ mount -o remount,rw /run/initramfs/cos-state
  ```
```

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 5]
```diff
@@ -398,3 +398,76 @@ If an upgrade is stuck in an `Upgrading System Service` state for an extended pe
...
+
+1. remove the `state.yaml`
+
+  ```
+  $ rm -f /run/initramfs/cos-state/state.yaml
+  ```
```
Comment (LucasSaintarbor) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:459]:
```suggestion
1. Remove the `state.yaml`.

  ```bash
  $ rm -f /run/initramfs/cos-state/state.yaml
  ```
```

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 6]
```diff
@@ -398,3 +398,76 @@ If an upgrade is stuck in an `Upgrading System Service` state for an extended pe
...
+
+1. remount the `state` partition to RO
+
+  ```
+  $ mount -o remount,ro /run/initramfs/cos-state
+  ```
```
Comment (LucasSaintarbor) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:465]:
```suggestion
1. Remount the `state` partition to RO.

  ```bash
  $ mount -o remount,ro /run/initramfs/cos-state
  ```
```

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 7]
```diff
@@ -398,3 +398,76 @@ If an upgrade is stuck in an `Upgrading System Service` state for an extended pe
...
+
+Then, you should pass post-draining with the next retry.
+
+- Related issue:
```
Comment (LucasSaintarbor) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:469]:
```suggestion
- Related issues:
```

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 8]
```diff
@@ -398,3 +398,76 @@ If an upgrade is stuck in an `Upgrading System Service` state for an extended pe
   - https://github.com/harvester/harvester/issues/3863#issuecomment-1539681311
 
 ---
+
+### 8. Upgrade stuck in the `Post-draining` state
+
+If you encounter the `post-draining` state like the following figure, that might be stuck in the OS upgrade process.
```
[the remark this answers is not in the document]
Comment (LucasSaintarbor) [reply docs/upgrade/v1-1-2-to-v1-2-0.md:404]: Making Post-draining bold since it shows up in the UI. Lmk if this is not correct, thanks!

**Review by LucasSaintarbor — COMMENTED** [review]
Made formatting and grammar changes + small NITs. Lmk if these changes are okay, thanks!

**Comment by Vicente-Cheng** [comment]
> Made formatting and grammar changes + small NITs. Lmk if these changes are okay, thanks!

Hi @LucasSaintarbor, thanks for the review.
Update all changes!

**Review by LucasSaintarbor — APPROVED** [review]
LGTM, thanks!

**Comment by bk201** [comment]
@Vicente-Cheng We also need to duplicate the file to the `version-1.2` directory.

**Comment by Vicente-Cheng** [comment]
> @Vicente-Cheng We also need to duplicate the file to the `version-1.2` directory.

Done! Thanks for the reminder!

**Review by bk201 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**